### PR TITLE
feat: oclif v2 migration

### PR DESCRIPTION
**Pull request by shrutiburman**
# Fixes # Issues in epic: [DII-393](https://issues.corp.twilio.com/browse/DII-393)

This PR:
 Merges all development done related to oclif-v2 migration into main. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [X] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [X] I have titled the PR appropriately
- [X] I have updated my branch with the main branch
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -57,16 +55,16 @@
...
-    "@oclif/test": "^1.2.6",
+    "@oclif/test": "^2.1.1",
     "@semantic-release/changelog": "^6.0.1",
     "@semantic-release/exec": "^6.0.3",
     "@semantic-release/git": "^10.0.1",
-    "@twilio/cli-test": "^2.1.0",
-    "aws-sdk": "^2.757.0",
-    "chai": "^4.2.0",
+    "@twilio/cli-test": "3.0.0",
+    "aws-sdk": "^2.443.0",
```
Comment (shrutiburman) [inline package.json:63]: aws-sdk is a mandatory dependency for oclif-dev, so the version is kept in [sync](https://github.com/oclif/dev-cli/blob/ef2ec22438e87fb7107b4c49d2c47584a2222817/package.json#L36) with oclif-dev.

File: .github/actions/build-rpm/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   "devDependencies": {
     "@types/node": "^16.4.10",
     "@vercel/ncc": "^0.25.1",
-    "typescript": "^4.0.5"
+    "typescript": "^4.0.5",
```
Comment (shrutiburman) [inline .github/actions/build-rpm/package.json:25]: Pipeline changes PR [464](https://github.com/twilio/twilio-cli/pull/464).

File: .github/scripts/pack_macos.sh [hunk 3]
```diff
@@ -47,10 +47,10 @@ pack_macos() {
 #  if [ "$REPOSITORY_OWNER" == "twilio" ]
 #  then
     import_certificate
-    npx oclif-dev pack:macos
+    npx oclif pack:macos
```
Comment (shrutiburman) [inline .github/scripts/pack_macos.sh:50]: [Pipeline changes details](https://github.com/twilio/twilio-cli/pull/464)

File: bin/dev [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env node
```
Comment (shrutiburman) [inline bin/dev:1]: [PR detail](https://github.com/twilio/twilio-cli/pull/459)

File: src/services/twilio-help/twilio-command-help.js [hunk 5]
```diff
@@ -1,38 +1,93 @@
 /* eslint-disable import/no-extraneous-dependencies */
-const CommandHelp = require('@oclif/plugin-help/lib/command.js');
-const list = require('@oclif/plugin-help/lib/list');
+const { CommandHelp } = require('@oclif/core');
```
Comment (shrutiburman) [inline src/services/twilio-help/twilio-command-help.js:2]: [PR Details](https://github.com/twilio/twilio-cli/pull/455)

**Review by charan678 — APPROVED** [review]
LGTM

**Pull request merged**